app/src/main/java/helium314/keyboard/latin/spellcheck/AndroidWordLevelSpellCheckerSession.java

```diff
@@ -47,6 +48,8 @@ public abstract class AndroidWordLevelSpellCheckerSession extends Session {
 
     public final static String[] EMPTY_STRING_ARRAY = new String[0];
 
+    public final static int FLAG_UNCHECKABLE = 0;
```

Comment:
This goes into `SuggestionsInfo` as `suggestionsAttributes`, but in the [documentation](https://developer.android.com/reference/android/view/textservice/SuggestionsInfo#getSuggestionsAttributes()) I can't find anything referring to _uncheckable_. 0 just seems to be when none of the `RESULT_ATTR_[...]` applies.

Reply:
ah I may have overlooked that, although a 0 flag is already used elsewhere in the code:
https://github.com/Helium314/HeliBoard/blame/136b45880e6c3aadb6d9e1b68f90851b1fbb94e7/app/src/main/java/helium314/keyboard/latin/spellcheck/AndroidSpellCheckerSession.java#L72
so it might need to be changed as well.
if we don't know what is the fallback behavior in the absence of flag then it would probably be wrong to name it like that with a constant

Reply:
The comment above says "_Neither RESULT_ATTR_IN_THE_DICTIONARY nor RESULT_ATTR_LOOKS_LIKE_TYPO_", and this also fits with documentation, which implies that 0 means _none of the flags_.
There is no information regarding 0 meaning "_uncheckable_".

Reply:
If you had spent more time to understand how the code works, you could have inferred that cached suggestions with `FLAG_UNCHECKABLE` just return `AndroidSpellCheckerService.getNotInDictEmptySuggestions(false)`, so this a flag of 0 is not returned to/handled by the native code. Essentially this constant could be named however we want as long as its purpose is clear.